waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+import numpy as np
+import os
+import zarr
+from tifffile import TiffFile
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.chnames = []
+
+        self.master_ome_tiff = self._get_master_ome_tiff(folder)
+        print(self.master_ome_tiff)
+        self._set_mm_meta()
```

Comment:
Add flag to extract data upon the initiation of the class? I found that this extra step can sometimes be a little annoying if I want to see all of the positions in the class without loading an array.  Default can be false.

```suggestion
    def __init__(self, folder, extract_data=False):
        """
        loads ome-tiff files in folder into zarr or numpy arrays
        Strategy:
            1. read all files in folder
            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict

        :param folder: folder containing all ome-tiff files
        :param reader: tifffile or aicsimageio
        """

        self._positions = {}
        self.mm_meta = None
        self.stage_positions = 0
        self.height = 0
        self.width = 0
        self.frames = 0
        self.slices = 0
        self.channels = 0
        self.chnames = []

        self.master_ome_tiff = self._get_master_ome_tiff(folder)
        print(self.master_ome_tiff)
        self._set_mm_meta()
        if extract_data:
            self._extract_data(self.master_ome_tiff)
        
```

Reply:
agreed, will implement this.